packages/gatsby-source-contentful/README.md

```diff
@@ -385,6 +388,44 @@ module.exports = {
 
 Check the [Reference Guide of gatsby-plugin-image](https://www.gatsbyjs.com/docs/reference/built-in-components/gatsby-plugin-image/) to get a deeper insight on how this works.
 
+### Building images on the fly via `useContentfulImage`
+
+With `useContentfulImage` and the URL to the image on the Contentful Image API you can create dynamic images on the fly:
+
+```js
+import { GatsbyImage } from "gatsby-plugin-image"
+import * as React from "react"
+import { useContentfulImage } from "gatsby-source-contentful/hooks"
+
+const MyComponent = () => {
+  const dynamicImage = useContentfulImage({
+    image: {
+      url: "//images.ctfassets.net/k8iqpp6u0ior/3BSI9CgDdAn1JchXmY5IJi/f97a2185b3395591b98008647ad6fd3c/camylla-battani-AoqgGAqrLpU-unsplash.jpg",
```

Comment:
What do you think about adding a note that variables are allowed as input values here? For some Gatsby hooks variables aren't allowed, so mentioning it explicitly could be beneficial for folks that are used to some of those hooks

Reply:
I'd rather go for the React standard that hooks are hooks. And hooks that get special processing aka don't support variables should be marked as such.


Without dynamic input data, this whole hook makes no sense to have :)